pkgs/tools/networking/v2ray/generic.nix

```diff
@@ -0,0 +1,53 @@
+{ lib, linkFarm, buildGoPackage, runCommand, makeWrapper, go
+
+# Version specific args
+, version, src, assets
+, ... }:
+
+let
+  assetsDrv = linkFarm "v2ray-assets" (lib.mapAttrsToList (name: path: {
+    inherit name path;
+  }) assets);
+
+  core = buildGoPackage rec {
```

Comment:
can you use [`buildGoModule`](https://nixos.org/nixpkgs/manual/#ssec-go-modules)?